tests/tlstest.py

```diff
@@ -1638,6 +1653,22 @@ def server_bind(self):
 
     test_no += 1
 
+    print("Test {0} - Heartbeat extension response callback".format(test_no))
+    synchro.send(b'R')
+    connection = connect()
+    settings = HandshakeSettings()
+    heartbeat_payload = os.urandom(100)
+    def heartbeat_response_check(message):
+        assert heartbeat_payload == message.payload
```

Comment:
if it receives a response, it should set a guard variable that is then checked after connection close to verify that the response was received

(but the check definitely has to be after testConnServer() as we need to do few reads and writes before we can be reasonably sure that the client had time to answer)

Reply:
How can I get variable out of callback scope? I'm trying something like this:
```
receive_payload = None
def heartbeat_response_check(message):
    received_payload = message.payload
```
But `received_payload` in and out are different. Is any simple way, how can I achieve that callback changes that variable?

Reply:
> How can I get variable out of callback scope?

either by using [`global`](https://docs.python.org/2.0/ref/global.html) or by changing a field in an object, not the object itself, compare:
```python
val = []
def fun():
   val[:] = ['some', 'values']
   print("ID of the object after assignment: {0}".format(id(val)))
print("ID of the object before assignment: {0}".format(id(val)))
print(val)
fun()
print(val)
```
with
```python
val = []
def fun():
   val = ['some', 'values']
   print("ID of the object after assignment: {0}".format(id(val)))
print("ID of the object before assignment: {0}".format(id(val)))
print(val)
fun()
print(val)
```

Reply:
Thanks, I didn't know that such a thing changes objects scope. 
Which of proposed solutions do you prefer here?